### ISSUE TITLE
Support QLoRA Training on AWQ Quantized Models

### ISSUE THREAD

**Issue by yhhtc1201** [huggingface/peft#2745]
### Feature request

QLoRA enables efficient fine-tuning of large models by combining low-bit quantization with LoRA layers. While BnB quantized models are supported, AWQ (Activation-aware Weight Quantization) models currently lack official support in QLoRA workflows. Could you add support for AWQ quantized models?

### Your contribution

.

**Comment by BenjaminBossan** [comment]
AWQ is supported, did you try it? If you need an example to get started, check the [AWQ tests](https://github.com/huggingface/peft/blob/04d41cbcd061bf1ab1185e111054bae012cb1894/tests/test_gpu_examples.py#L3491).

**Comment by yhhtc1201** [comment]
The testing function works successfully, but it doesn't seem to be applicable to all AWQ models. For example, when I tested Gemma-3-27B-awq, I encountered a gradient nan issue.
https://huggingface.co/gaunernst/gemma-3-27b-it-int4-awq

**Comment by BenjaminBossan** [comment]
@yhhtc1201 This is hard for us to investigate with the given information. Could you please share the training script you're using that results in gradients being nan? If this contains some private data, feel free to substitute for a public dataset. Please also share how you're doing training (single GPU, distributed, etc.), as well as the versions of libraries you use (PEFT, transformers, torch, AWQ).

**Comment by yhhtc1201** [comment]
I just replaced the test model in the [AWQ tests](https://github.com/huggingface/peft/blob/04d41cbcd061bf1ab1185e111054bae012cb1894/tests/test_gpu_examples.py#L3491) with gemma-3-27b-it-int4-awq and encountered the nan issue.
Model link: https://huggingface.co/gaunernst/gemma-3-27b-it-int4-awq

**Comment by BenjaminBossan** [comment]
As I mentioned, the information you provide is not sufficient to investigate the issue. Please provide a small script that allows us to replicate the issue. Also, notice that [AWQ is no longer being maintained](https://github.com/casper-hansen/AutoAWQ/blob/88e4c76b20755db275574e6a03c83c84ba3bece5/README.md), so it's recommended to switch to another quantization method.

**Comment by benjamin-marie** [comment]
> As I mentioned, the information you provide is not sufficient to investigate the issue. Please provide a small script that allows us to replicate the issue. Also, notice that [AWQ is no longer being maintained](https://github.com/casper-hansen/AutoAWQ/blob/88e4c76b20755db275574e6a03c83c84ba3bece5/README.md), so it's recommended to switch to another quantization method.

AutoAWQ is no longer maintained but the AWQ format still lives. Quantization frameworks, e.g., AutoRound and LLM Compressor, now propose to export models with the AWQ format.

**Comment by BenjaminBossan** [comment]
> AutoAWQ is no longer maintained but the AWQ format still lives. Quantization frameworks, e.g., AutoRound and LLM Compressor, now propose to export models with the AWQ format.

Since we still rely on the `awq` package for this, compatibility could still sooner or later break, even if the format lives on. I guess we have to monitor the development and switch the package when this becomes necessary, but at this stage it's too early to say.

**Comment by benjamin-marie** [comment]
GPTQModel is working on adding AWQ support as well. Since GPTQModel is already supported by Transformers, this could be the easiest path to preserve compatibility with AWQ. 
https://github.com/ModelCloud/GPTQModel/pull/1703

**Comment by BenjaminBossan** [comment]
> GPTQModel is working on adding AWQ support as well.

Nice, I wasn't aware of that, thanks for letting me know. Since we already integrate with gptqmodel, this sounds like the best candidate to eventually make the switch.
